adafruit_debouncer.py

```diff
@@ -31,23 +31,34 @@
 from micropython import const
 from adafruit_ticks import ticks_ms, ticks_diff
 
-_DEBOUNCED_STATE = const(0x01)
-_UNSTABLE_STATE = const(0x02)
-_CHANGED_STATE = const(0x04)
+try:
+    from typing import Callable, Optional, Union
+except ImportError:
+    pass
+from digitalio import DigitalInOut
 
-_TICKS_PER_SEC = const(1000)
+_DEBOUNCED_STATE: int = const(0x01)
+_UNSTABLE_STATE: int = const(0x02)
+_CHANGED_STATE: int = const(0x04)
+
+_TICKS_PER_SEC: int = const(1000)
 
 
 class Debouncer:
     """Debounce an input pin or an arbitrary predicate"""
 
-    def __init__(self, io_or_predicate, interval=0.010):
+    def __init__(
+        self,
+        io_or_predicate: Union[DigitalInOut, Callable[[], bool]],
+        interval: float = 0.010,
+    ) -> None:
         """Make an instance.
-        :param DigitalInOut/function io_or_predicate: the DigitalIO or function to debounce
-        :param int interval: bounce threshold in seconds (default is 0.010, i.e. 10 milliseconds)
+        :param DigitalInOut/function io_or_predicate: the DigitalIO or
+                                                      function that returns a boolean to debounce
+        :param float interval: bounce threshold in seconds (default is 0.010, i.e. 10 milliseconds)
         """
         self.state = 0x00
-        if hasattr(io_or_predicate, "value"):
+        if isinstance(io_or_predicate, DigitalInOut):
```

Comment:
Hmm, I am also wondering about this.  As is, if you intended to use this library only with functions, you'd never need to import `digitalio`, saving some RAM.  With this change, you do.  I think this is a good change, but also looking for a second opinion on this @FoamyGuy!